src/ray/rpc/metrics_agent_client.h

```diff
@@ -40,6 +40,19 @@ class MetricsAgentClient {
   /// \param[in] request The request message.
   /// \param[in] callback The callback function that handles reply.
   VOID_RPC_CLIENT_VIRTUAL_METHOD_DECL(ReporterService, ReportOCMetrics)
+
+  /// Send a health check request to the metrics agent.
+  ///
+  /// \param[in] request The request message.
+  /// \param[in] callback The callback function that handles reply.
+  VOID_RPC_CLIENT_VIRTUAL_METHOD_DECL(ReporterService, HealthCheck)
+
+  /// Initialize an exporter (e.g. metrics, events exporter).
+  ///
+  /// This function ensures that the server is ready to receive metrics before
+  /// initializing the exporter. If the server is not ready, it will retry for
+  /// a number of times.
+  virtual void InitExporter(std::function<void()> init_exporter_fn) = 0;
 };
 
 class MetricsAgentClientImpl : public MetricsAgentClient {
```

Comment:
nit: let's standardize on `FooBarInterface` for virtual class and `FooBar` for concrete implementation (that's what we have mostly)

can you rename in a separate PR sometime?